session/src/main/scala/akka/cassandra/session/SessionProvider.scala

```diff
@@ -2,15 +2,14 @@
  * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
  */
 
-package akka.persistence.cassandra
+package akka.cassandra.session
 
-import scala.collection.immutable
-import scala.concurrent.Future
+import akka.actor.{ ActorSystem, ExtendedActorSystem }
 import com.datastax.driver.core.Session
 import com.typesafe.config.Config
-import scala.concurrent.ExecutionContext
-import akka.actor.ExtendedActorSystem
-import akka.actor.ActorSystem
+
```

Comment:
Do we need scalads/javadsl for `SessionProvider`, it has the single method `connect` returning `Future`. Perhaps not needed because it's not intended to be used or implemented by end users.

Reply:
I think some end users will implement `SessionProvider`. For example, we have had a customer request to be able to use [`DseSession`](https://docs.datastax.com/en/drivers/java-dse/2.0/com/datastax/dse/driver/api/core/DseSession.html) from the commercial DataStax driver (which extends the OSS [`Session`](https://docs.datastax.com/en/drivers/java/4.0/com/datastax/oss/driver/api/core/session/Session.html?is-external=true) interface). I've talked with @ennru about making this a documented extension point.